File: app/src/main/java/fr/gstraymond/ui/view/impl/CostView.java
package fr.gstraymond.ui.view.impl;

import android.widget.TextView;

import com.magic.card.search.commons.log.Log;

import fr.gstraymond.R;
import fr.gstraymond.models.response.Card;
import fr.gstraymond.models.response.Publication;
import fr.gstraymond.ui.view.CommonDisplayableView;

public class CostView extends CommonDisplayableView<TextView> {

    private static double max$ = 1d;
    private static double max$$ = 10d;

    private Log log = new Log(this);

    @Override
    public boolean display(Card card) {
        return true;
    }

    @Override
    public int getId() {
        return R.id.array_adapter_card_price;
    }

    @Override
    public void setValue(Card card, int position) {
        int count$ = 0;
        int count$$ = 0;
        int count$$$ = 0;
        for (Publication publication : card.getPublications()) {
            if (publication.getPrice() > max$$) count$$$++;
            else if (publication.getPrice() > max$) count$$++;
            else if (publication.getPrice() > 0d) count$++;
        }

        String $$$ = "?";
        if (count$ > 0) $$$ = "$";
        else if (count$$ > 0) $$$ = "$$";
        else if (count$$$ > 0) $$$ = "$$$";

        log.d("$ %s $$ %s $$$ %s -> %s", count$, count$$, count$$$, $$$);

        if ($$$.equals("?")) {
            display(false);
        } else {
            display(true);
            getView().setText($$$);
        }
    }
}

File: app/src/main/kotlin/fr/gstraymond/ui/view/impl/CostView.kt
package fr.gstraymond.ui.view.impl

import android.content.Context
import android.widget.ImageView
import fr.gstraymond.R
import fr.gstraymond.models.search.response.Card
import fr.gstraymond.ui.view.CommonDisplayableView
import fr.gstraymond.utils.color

class CostView(private val context: Context) : CommonDisplayableView<ImageView>(R.id.array_adapter_card_price) {

    private val max1 = 1.0
    private val max2 = 10.0

    override fun display(view: ImageView, card: Card) = card.publications.any { it.price > 0 }

    override fun setValue(view: ImageView, card: Card, position: Int) {
        val prices = card.publications.map { it.price }.filter { it > 0 }

        val color = when {
            prices.any { it <= max1 } -> R.color.colorPrimaryDark
            prices.any { it > max1 && it <= max2 } -> R.color.colorPrimary
            else -> R.color.colorAccent
        }

        view.setColorFilter(context.resources.color(color))
    }
}
